### PR TITLE
Improved Kover documentation

### LINKED ISSUE #282: Improve Kover documentation
Proposed changes:
- add a detailed description of the Kotlin Kover: purpose, differences, pros and cons.
- explicitly mention example projects
- improve the description of the minimum setup for the ”quick start”
- explain the difference between Kotlin/Groovy code snippets in more detail
- add the requirement to enable Kover in all modules in the description
- merge “Single-project tasks” with “Configuring project” and “Cross-project coverage with Configuring merged reports”
- describe the difference between kover{} and kovermerged{}
- add a Kover HTML report description
    - when reading the report - some of the lines are green and some are red. It’s quite straightforward - they’re either fully tested or not at all. But the yellow ones are a bit unclear
    - explicitly indicate that the report only shows coverage by lines, it cannot provide detailed information on each expression in a line

### PR DESCRIPTION
**Pull request by shanshin**
Resolves #282

### CODE REVIEW

File: docs/gradle-plugin/index.md [hunk 1]
```diff
@@ -23,10 +27,12 @@
 
 * Collection of code coverage through `JVM` tests (JS and native targets are not supported yet).
 * generating `HTML` and `XML` reports.
+* Verification rules with bounds to keep track of coverage.
+* Create tasks to generate reports and automatically adjusts the dependencies between it
```
Comment (sandwwraith) [inline docs/gradle-plugin/index.md:31]:
```suggestion
* Create tasks to generate reports and automatically adjust the dependencies between it
```

To be consistent with the rest of the list. IMO, everything here should be in third-person, but that depends on the style

File: docs/gradle-plugin/index.md [hunk 2]
```diff
@@ -154,8 +160,8 @@ koverReport {
 ```
 
 ## Multiproject build
-[Multi-project build](https://docs.gradle.org/current/userguide/multi_project_builds.html#sec:creating_multi_project_builds) (sometimes called multimodule project) 
-- this is a Gradle build in which there are several Gradle projects (most often each of which has its own `build.gradle` or `build.gradle.kts` file)
+[Multi-project build](https://docs.gradle.org/current/userguide/multi_project_builds.html#sec:creating_multi_project_builds) 
+(sometimes called multimodule project) - this is a Gradle build in which there are several Gradle projects (most often each of which has its own `build.gradle` or `build.gradle.kts` file)
```
Comment (sandwwraith) [inline docs/gradle-plugin/index.md:164]:
```suggestion
(sometimes called multimodule project) - this is a Gradle build in which there are several Gradle projects (most often each has its own `build.gradle` or `build.gradle.kts` file)
```

File: docs/gradle-plugin/index.md [hunk 3]
```diff
@@ -241,6 +249,16 @@ Example:
...
 
 
+## HTML report description
+![Example of HTML report](html.png)
+- green indicates a line that has been executed at least once (covered)
+- red indicates a line that has never been executed (missed)
+- yellow indicates a line in which there is a branching, but at least one of the branches was not executed (partially covered)
+- code that was excluded from the report using filters is not highlighted in any color
+
+In the HTML report, Kover shows coverage by lines, it cannot display detailed coverage for an expression or separately for each code branch on a line.
```
Comment (sandwwraith) [inline docs/gradle-plugin/index.md:259]: This is the limitation of default Kover tool, but Jacoco can do this?

File: docs/gradle-plugin/index.md [hunk 4]
```diff
@@ -1,4 +1,7 @@
 # Kover Gradle Plugin
+
+Gradle plugin that measures the coverage with tests of code running on the JVM and generates coverage reports.
```
Comment (sandwwraith) [inline docs/gradle-plugin/index.md:3]:
```suggestion
Gradle plugin that measures the coverage for tests running on the JVM and generates coverage reports.
```

'tests of code' sounds too wordy

**Review by sandwwraith — CHANGES_REQUESTED** [review]

File: docs/gradle-plugin/index.md (hunk 3 above)
Comment (shanshin) [reply docs/gradle-plugin/index.md:259]: No, JaCoCo also can't

**Review by sandwwraith — APPROVED** [review]
